# <b>Folium Exercise</b>



by : Tito Tamaro <br>
Source Dataframe : https://www.kaggle.com/kveykva/code    
Source Picture : https://pokemondb.net/pokedex/national

In [2]:
import folium
import numpy as np
import pandas as pd
import random
from bs4 import BeautifulSoup
import requests
%matplotlib inline

# Pokemon Picture Extraction

In [3]:
url = 'https://pokemondb.net/pokedex/national'
data = requests.get(url)
soup = BeautifulSoup (data.text, 'html.parser')

storage_img = []
for img in soup.find_all('span',{'class':'img-fixed img-sprite'}):
    storage_img.append(img['data-src'])

# EDA - Pokemon

In [4]:
df = pd.read_csv('pokemon-spawns.csv')

In [5]:
df.isna().sum()

s2_id           0
s2_token        0
num             0
name            0
lat             0
lng             0
encounter_ms    0
disppear_ms     0
dtype: int64

In [6]:
df.head()

s2_id     s2_token  num     name        lat         lng  \
0 -9185794522947256000  8085808cc6d   13   Weedle  37.793592 -122.408721   
1 -9185794529389707000  8085808b51d   16   Pidgey  37.794746 -122.406420   
2 -9185794529389707000  8085808b271   41    Zubat  37.794999 -122.404384   
3 -9185794082713108000  808580f3587   16   Pidgey  37.795644 -122.407128   
4 -9185794076270658000  808580f4b1d   60  Poliwag  37.795592 -122.406331   

    encounter_ms    disppear_ms  
0  1469520187732  1469519919988  
1  1469520297172  1469519919992  
2  1469520709924  1469519919991  
3             -1  1469519920134  
4  1469520741876  1469519920153

In [7]:
df.tail()

s2_id     s2_token  num         name        lat  \
314100 -9185787356794323000  8085871158f   84        Doduo  37.778626   
314101 -9185786935887528000  80858773019   56       Mankey  37.769260   
314102 -9185786940182495000  80858772545   29  Nidoran (f)  37.768922   
314103 -9185787373974192000  8085870d1db   10     Caterpie  37.773062   
314104 -9185786976689717000  80858769d49   79     Slowpoke  37.768654   

               lng   encounter_ms    disppear_ms  
314100 -122.475958  1469670899836  1469670453557  
314101 -122.483652  1469671114084  1469670453557  
314102 -122.483298  1469670572436  1469670453557  
314103 -122.483387  1469671257676  1469670453556  
314104 -122.469678  1469671286340  1469670453557

In [8]:
df.shape

(314105, 8)

Tugas : memunculkan minimal 1500 pokemon per-run folium, dengan custom Icon

In [9]:
df['name'].nunique()

140

Walaupun data terdiri dari 314105 baris, namun jumlah unique name pokemon adalah 140

# Mapping 

Mapping dititikberatkan pada area San Fransisco Bay

In [73]:
m = folium.Map(
location = [37.801481, -122.460002],
tiles='OpenStreetMap',
zoom_start = 13
)

for i in range(0, 1501):
    num = random.randint(0,len(df))
    folium.Marker(
    [df.iloc[num]['lat'], df.iloc[num]['lng']],
    popup = f'{df.iloc[num]["name"]}',
    tooltip =  f'<img width="100px" src="https://img.pokemondb.net/sprites/sword-shield/icon/{df.iloc[num]["name"].lower()}.png"/><center>{df.iloc[num]["name"]}</center>',
    icon = folium.features.CustomIcon(f'https://img.pokemondb.net/sprites/sword-shield/icon/{df.iloc[num]["name"].lower()}.png', icon_size=(50,50))).add_to(m)

m